kpops/components/base_components/pipeline_component.py

```diff
@@ -67,6 +71,25 @@ def __init__(self, **kwargs) -> None:
         self.set_input_topics()
         self.set_output_topics()
 
+    def get_input_topics(self) -> list[str]:
+        """Get all the input topics from config."""
+        return []
+
+    def get_extra_input_topics(self) -> dict[str, list[str]]:
+        """Get extra input topics list from config."""
+        return {}
+
+    def get_output_topic(self) -> str | None:
+        """Get output topic from config."""
+
+    def get_extra_output_topics(self) -> dict[str, str]:
+        """Get extra output topics list from config."""
+        return {}
```

Comment:
how about we make these properties?

Reply:
But this should be possible to override. Maybe is not clear because it doesn't have the override keyword. Should I put it ?

Reply:
`@property` can be overridden, too